File: Backend/Logic/Data/Idea.cs
/** 
 *  Idea.cs
 *  --------------------
 *  Represents a Idea object from the database.
 *  This file contains the data for the Idea
 *  ---------------------------------------------------------
 *   Author: Mr. Roland Fehr
 *   Last modified: 28.10.2023
 *   Version: 1.0
*/

namespace Logic.Data
{
    /// <summary>
    ///   This class contains data for the Idea
    /// </summary>
    public class Idea
    {
        public string Id { get; set; }
        public string Thought { get; set; }
        public int Likes { get; set; }
        public int Dislikes { get; set; }
    }
}

File: FrontEnd/src/models/Idea.ts
/*
 * Idea.ts
 * -----------------------------
 * This file is the model for the Idea object.
 * ----------------------------------------------------------
 * Author:  Mr. Roland Fehr
 * Date Created:  01/12/2023
 * Last Modified: 01/12/2023
 * Version: 0.0.1
*/

export default interface Idea{
    id: string;
    thought: string;
    likes: number;
    dislikes: number;
}